CMakeLists.txt

```diff
@@ -641,6 +642,28 @@ if (BUILD_TESTING)
     endif()
 endif()
 
+if (S2N_INTEG_TESTS)
+    find_package (Python3 COMPONENTS Interpreter Development)
+    file(GLOB integv2_test_files "${PROJECT_SOURCE_DIR}/tests/integrationv2/test_*.py")
+    foreach(test_file_path ${integv2_test_files})
+        get_filename_component(test_filename ${test_file_path} NAME_WE)
+        string(REGEX REPLACE "^test_" "" test_name ${test_filename})
+        string(CONCAT test_target "integ_" ${test_name})
+
+        add_custom_target(${test_target}
+            COMMAND
+                ${CMAKE_COMMAND} -E env
+                DYLD_LIBRARY_PATH="${PROJECT_SOURCE_DIR}/libcrypto-root/lib":$ENV{DYLD_LIBRARY_PATH}
+                LD_LIBRARY_PATH="${PROJECT_SOURCE_DIR}/libcrypto-root/lib":"${PROJECT_SOURCE_DIR}/test-deps/openssl-1.1.1/lib":"${PROJECT_SOURCE_DIR}/test-deps/gnutls37/nettle/lib":$ENV{LD_LIBRARY_PATH}
+                PATH="${PROJECT_SOURCE_DIR}/bin":"${PROJECT_SOURCE_DIR}/test-deps/openssl-1.1.1/bin":"${PROJECT_SOURCE_DIR}/test-deps/gnutls37/bin":$ENV{PATH}
+                PYTHONNOUSERSITE=1
+                S2N_INTEG_TEST=1
```

Comment:
Out of the scope of this PR, but I'll also throw out my unsolicited opinion that using special code paths for _all_ integration tests scares me.

This is probably a bit paranoid, but it looks like setting this environment variable let's you call `s2n_config_set_unsafe_for_testing`. If we were to add a regression that resulted in a production code path calling `s2n_config_set_unsafe` I don't think we wouldn't catch that in unit or integ tests, because it's explicitly allowed for all of them.

As a future solution maybe we could only set `S2N_INTEG_TEST=1` on the tests that actually need it? That could probably be driven from the python layer.